### PR TITLE
Add recipe for symopt

### PR DESCRIPTION
**Pull request by spcornelius**
Checklist

- [ x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ x] Source is from official source
- [ x] Package does not vendor other packages
- [ x] Build number is 0
- [ x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/symopt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - numpy
+    - sympy
+    - scipy
+    - cyipopt
+    - orderedset
+
+test:
+  imports:
+    - {{ name }}
+  requires:
```
Comment (marcelotrevisani) [inline recipes/symopt/meta.yaml]: that is odd
do you need ``pytest`` and ``fortran`` ? You can drop it, however, if you need it to run this project you should add it as ``run`` dependency

Comment (spcornelius) [reply recipes/symopt/meta.yaml]: Hi Marcelo. The reason for the above arrangement is that one of the available SymPy backends for code generation requires fortran. However, there are other such backends (including the default setting in this package) that do not require fortran. As such, a user could perfectly well install and run this without fortran, but fortran is strictly required to run (all of) tests.

Does that make sense?

File: recipes/symopt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: 24335e1ed3306a5b30f9f7eab97b8bb66b562c625a260c32d0f8caaf2a485ec9
+
+build:
+  skip: True  # [win]
+  number: 0
+  script: python -m pip install . --no-deps -vv
+
+requirements:
+  build:
+    - python >= 3.6
```
Comment (chrisburr) [inline recipes/symopt/meta.yaml]: There should be a space before the version constraint but not before the number:
```suggestion
    - python >=3.6
```
(`conda-build` doesn't enforce this everywhere so builds work like this by feedstock creation doesn't)

File: recipes/symopt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - setuptools
+    - pip
+  run:
+    - python >= 3.6
```
Comment (chrisburr) [inline recipes/symopt/meta.yaml]: Same
```suggestion
    - python >=3.6
```

**Pull request merged**